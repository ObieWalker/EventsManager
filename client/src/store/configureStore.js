import { createStore, applyMiddleware } from 'redux';
import thunkMiddleware from 'redux-thunk';
import { createLogger } from 'redux-logger';
import rootReducer from '../reducers/rootReducer';

const middleware = applyMiddleware(thunkMiddleware, createLogger({ collapsed: true }));

const Store = createStore(
    rootReducer,
    applyMiddleware(middleware)
);

export default Store;