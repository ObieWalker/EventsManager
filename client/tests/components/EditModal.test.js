import React from 'react';
import expect from 'expect';
import { shallow } from 'enzyme';
import configureStore from 'redux-mock-store';
import connect, { EditModal }
  from '../../src/components/UserDashboard/EditModal';
import { events } from '../__mocks__/eventsData';

let mountedComponent;
let props;
const mockStore = configureStore();
// const createCenter = jest.fn(() => Promise.resolve({}));

let wrapper;
const handleClose = jest.fn();
const hideModal = jest.fn();
const updateSuccess = '';
const updateError = '';
const event = events.event2;
const editEvent = jest.fn(() => Promise.resolve({}));
const getComponent = () => {
  if (!mountedComponent) {
    props = {
      event,
      handleClose,
      editEvent,
      updateSuccess,
      updateError,
      hideModal
    };
    mountedComponent = shallow(<EditModal {...props} />);
  }
  return mountedComponent;
};

describe('Modify Center Component', () => {
  beforeEach(() => {});
  wrapper = getComponent();

  it('component successfully rendered', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should have a method that handles change to each input element', () => {
    expect(wrapper.instance().onSubmit).toBeDefined();
  });
  it('should have a method that handles change to each input element', () => {
    expect(wrapper.instance().onSliderChange).toBeDefined();
  });
  it('should have a method that handles change to each input element', () => {
    expect(wrapper.instance().formIsValid).toBeDefined();
  });
  it('should have a method that handles change to each input element', () => {
    expect(wrapper.instance().handleDateChange).toBeDefined();
  });
  it('testing create center function', () => {
    wrapper = getComponent();
    const e = { preventDefault: () => undefined };
    e.preventDefault = jest.fn();
    wrapper.instance().onSubmit(e);
    expect(e.preventDefault).toHaveBeenCalled();
  });
//   it('testing clear function', () => {
//     wrapper = getComponent();
//     wrapper.instance().reset();
//     expect(wrapper.state().name).toEqual('');
//     expect(wrapper.state().address).toEqual('');
//     expect(wrapper.state().city).toEqual('');
//     expect(wrapper.state().capacity).toEqual('');
//     expect(wrapper.state().facility).toEqual('');
//     expect(wrapper.state().errors).toEqual({});
//   });
});

describe('Connected Component', () => {
  describe('Connected mountedLogin', () => {
    it('component successfully rendered', () => {
      const store = mockStore({
        event,
        handleClose,
        editEvent,
        updateSuccess,
        updateError,
        hideModal
      });
      wrapper = shallow(<connect store={store}
        event={event}
        updateSuccess={updateSuccess}
        updateError={updateError}
        editEvent={editEvent}
        hideModal={hideModal}
        handleClose={handleClose}
      />);
      expect(wrapper.length).toBe(1);
    });
  });
});
