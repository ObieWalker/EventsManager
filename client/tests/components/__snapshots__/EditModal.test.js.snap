// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Modify Center Component component successfully rendered 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <EditModal
    editEvent={[MockFunction]}
    event={
      Object {
        "Center": Object {
          "address": "23, Segun Samuel Street",
          "city": "Denrele",
          "image": "",
          "name": "The Hot Abode",
        },
        "centerId": 6,
        "createdAt": "2018-07-24T16:16:11.648Z",
        "date": "2018-9-04",
        "eventType": "Meeting",
        "guestNo": 20,
        "id": 4,
        "isCancelled": null,
        "updatedAt": "2018-07-24T16:16:11.648Z",
        "userId": 5,
      }
    }
    handleClose={[MockFunction]}
    hideModal={[MockFunction]}
    updateError=""
    updateSuccess=""
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "dateDefaultValue": "2018-9-04",
      "dateOnChange": [Function],
      "dateValue": undefined,
      "errors": Object {},
      "event": Object {
        "Center": Object {
          "address": "23, Segun Samuel Street",
          "city": "Denrele",
          "image": "",
          "name": "The Hot Abode",
        },
        "centerId": 6,
        "createdAt": "2018-07-24T16:16:11.648Z",
        "date": "2018-9-04",
        "eventType": "Meeting",
        "guestNo": 20,
        "id": 4,
        "isCancelled": null,
        "updatedAt": "2018-07-24T16:16:11.648Z",
        "userId": 5,
      },
      "eventCenter": "The Hot Abode",
      "eventTypeDefaultValue": "Meeting",
      "eventTypeValue": undefined,
      "guestDefaultValue": 20,
      "guestValue": 20,
      "onAfterChange": [Function],
      "onChange": undefined,
      "onFocus": undefined,
      "sliderMax": 100000,
      "sliderOnChange": [Function],
      "sliderStep": 50,
      "submitOnClick": [Function],
    },
    "ref": null,
    "rendered": null,
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "dateDefaultValue": "2018-9-04",
        "dateOnChange": [Function],
        "dateValue": undefined,
        "errors": Object {},
        "event": Object {
          "Center": Object {
            "address": "23, Segun Samuel Street",
            "city": "Denrele",
            "image": "",
            "name": "The Hot Abode",
          },
          "centerId": 6,
          "createdAt": "2018-07-24T16:16:11.648Z",
          "date": "2018-9-04",
          "eventType": "Meeting",
          "guestNo": 20,
          "id": 4,
          "isCancelled": null,
          "updatedAt": "2018-07-24T16:16:11.648Z",
          "userId": 5,
        },
        "eventCenter": "The Hot Abode",
        "eventTypeDefaultValue": "Meeting",
        "eventTypeValue": undefined,
        "guestDefaultValue": 20,
        "guestValue": 20,
        "onAfterChange": [Function],
        "onChange": undefined,
        "onFocus": undefined,
        "sliderMax": 100000,
        "sliderOnChange": [Function],
        "sliderStep": 50,
        "submitOnClick": [Function],
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
